design/batch_retrieval_design.md

```diff
@@ -288,7 +288,31 @@ The following integration tests will be implemented in Secrets Provider for K8s
 
 ## Performance tests
 
---
+The following performance tests will be run using *Jmeter*.
+These tests will be run for both the new API and the existing one, and compare the results between them.
+
+#### Benchmark
+
+| Section     | Parameter                      | Value          |
+| ----------- | ------------------------------ | -------------- |
+| Conjur      | \# of standbys                 | 2              |
+| Conjur      | \# of followers                | 1              |
+| Environment | Kubernetes                     | OpenShift 3.11 |
+| Environment | \# of cores                    | 16             |
+| Environment | \# of nodes                    | 1              |
+| Test        | \# of requesting threads       | 20             |
+| Test        | \# of variables                | 1025           |
+| Test        | \# of variabels per policy     | 5              |
+| Test        | \# of policies                 | 205            |
+| Test        | Variable ID length             | 50             |
+| Test        | Variable value length (secret) | 50             |
+
+
+
+|      | Scenario      | Given                                                        | When                                                         | Then                                                         | Manual/Automated |
+| ---- | ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ---------------- |
+| 1    | Response time | - Conjur is running<br />- Host is defined<br />- 1024 Secrets are defined<br /> | For X=1, X<=1024, X=X\*2:<br />    Concurrently retrieve X secrets for 5 minutes, measure average response time.<br />    Repeat each step 3 times.<br /> | Calculate the average response time between the 3 cycles for each step.<br />Plot the average response time per number of secrets provider. | Manual           |
+| 2    | Throughput    | - Conjur is running<br />- Host is defined<br />- 1024 Secrets are defined<br /> | For X=1, X<=1024, X=X\*2:<br />    Concurrently retrieve X secrets for 5 minutes, measure throughput.<br />    Repeat each step 3 times.<br /> | Calculate the throughput between the 3 cycles for each step.<br />Plot the throughput per number of secrets provider. | Manual           |
```

Comment:
so we will be testing 1, 1024, 512 secrets in the body of batch request?

Reply:
We will be testing 1,2,4,8,16,32,64,128,256,512,1024 secrets in the body. each as its own test, and plot the results.

Reply:
That is not clear here. It seems to me that we are only running with 3 types of X values (1, 1024, 512 secrets). Please highlight that. It is imperative we are clear about what we are testing otherwise it will lead to misunderstandings and delay